### ISSUE TITLE
Bug - Duplicate author roles created due to auto-save timing issue. 

### ISSUE THREAD

**Issue by tmcelrath** [SpeciesFileGroup/taxonworks#1112]
**Describe the bug**
Duplicate author roles created due to auto-save timing issue. 

**To Reproduce**
1. When I am in the "Edit Taxon Name" task.
2. Then I select a source or verbatim date, or verbatim year. 
3. And I quickly go to the "Person" sub-tab under Author and click "Clone from Source" before the auto-save completes.
4. Then two people are created and listed from the "Clone from Source" button. This may be latency based, as it is difficult to replicate. 

**Expected behavior**
Should not duplicate authors

**Environment (please identify where you experience this bug:**
- Production
- Chrome

**Comment by mjy** [comment]
YAPWAS - yet another problem with auto-save.

**Comment by jlpereira** [comment]
Already fixed

**Comment by proceps** [comment]
People are still duplicated occasionally. I do not know if this is the same combination of conditions. I cannot find any specific pattern, but about one of 10-20 records I experience duplication.

**Comment by tmcelrath** [comment]
Agreed. Got this again today.

**Comment by mjy** [comment]
@tmcelrath Still exists?

**Comment by tmcelrath** [comment]
Not sure, haven't been adding lots of authors lately.

**Comment by proceps** [comment]
It is. And it is not related to autosave. I saw this behavior in various places. Source authors, sometimes get duplicated. I got asserted distribution duplicate (with the same citation, from a single save). In all cases it looks the same. You click the save button. The form get stuck, few seconds letter you have it save with the response duplicate. It looks like the database get pinged several time, and when response come through, it save the same record multiple times.

**Comment by mjy** [comment]
@proceps we need to identify specific places in the interface now.  Which
task/url.

**Comment by proceps** [comment]
I saw this in New Source (author role and editor role)
I saw this in New Asserted distribution task (not radial). One time it created me 3 identical records. I did not check if it is only a duplicated citation or duplicated OTU/Area record, I just deleted those.
And I cannot replicate those. Those things just happen from time to time. Not often. What I noticed, sometime the web site slows down (everywhere). I told you one day. I do not know what it is, but when the web site slow, chances of duplication are higher.
